src/main/java/org/spectra/cluster/model/consensus/GreedyConsensusSpectrum.java

```diff
@@ -341,4 +413,44 @@ public long getSummedPrecursorMz() {
     public int getSummedCharge() {
         return sumCharge;
     }
+
+    @Override
+    public int getPrecursorMz() {
+        return averagePrecursorMz;
+    }
+
+    @Override
+    public int getPrecursorCharge() {
+        return averageCharge;
+    }
+
+    @Override
+    public int[] getMzVector() {
+        return Arrays.stream(consensusPeaks).mapToInt(BinaryPeak::getMz).toArray();
```

Comment:
don't we have to check `isDirty()` first and regenerate the consensus spectrum if needed?